src/crypto/algorithms/megolm.js

```diff
@@ -527,6 +527,10 @@ utils.inherits(MegolmDecryption, base.DecryptionAlgorithm);
  *   problem decrypting the event
  */
 MegolmDecryption.prototype.decryptEvent = function(event) {
+    this._decryptEvent(event, true);
+};
+
+MegolmDecryption.prototype._decryptEvent = function(event, requestKeysOnFail) {
```

Comment:
I understand this is an internal method, but I'd like there to be a bit of prose by what you mean by "fail" on `requestKeysOnFail`. I see this exclusively means if the error message is `OLM.UNKNOWN_MESSAGE_INDEX` but that's as clear as mud.